lib/common.js

```diff
@@ -545,6 +574,7 @@ exports.readMysqlValue = function(parser, column, columnSchema) {
     case MysqlTypes.DECIMAL: // Deprecated in MySQL > 5.0.3
     case MysqlTypes.NEWDATE: // Not used
     default:
+      // TODO: Remove throw statement to make module safe
```

Comment:
Hmm interesting about performance, hadn't thought of that. If we drop the `throw` here and check for a `instanceof Error` in `lib/row_events.js` we could have the same effect though.